### PR TITLE
[MRG]Improve pip install

### LINKED ISSUE #279: Enable to install packages with GPU support
#### Software and Hardware Versions 
ModelCI v1.x.x, CUDA Version vx.x.x, GPU device used...

#### Problem description

Currently, modelci does not support installing Cuda version related packages, especially PyTorch and TensorFlow with GPU support

#### Other Information 
Things you tried, stack traces, related issues, suggestions on how to fix it...

### PR DESCRIPTION
**Pull request by univerone**
Closes #279 

#### Before submitting this PR, please make sure you have:

- [x] run `python -m pytest tests/` and confirmed all checks still pass.
- [x] verified that any code or assets from external sources are properly credited.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -5,55 +5,32 @@
...
-
-if sys.version_info < (3,):
-    print('Python 2 has reached end-of-life and is no longer supported by PyTorch.')
-    sys.exit(-1)
-if sys.platform != 'linux':
-    TRITON_CLIENT_INSTALL = False
-else:
-    TRITON_CLIENT_INSTALL = True
-
-python_min_version = (3, 7, 0)
```
Comment (YuanmingLeee) [inline setup.py:32]: Does this pip installation check the Python version? It is not able to run on versions below 3.7

File: setup.py [hunk 2]
```diff
@@ -79,17 +57,82 @@ def install_triton_client():
...
+torch_cuda_version = "cpu"
+tensorflow_cuda_version = "cpu"
+
+
+if "CUDA_HOME" in os.environ:
+    cuda_version_file = os.path.join(os.environ["CUDA_HOME"], "version.txt")
+    if os.path.exists(cuda_version_file):
+        with open(cuda_version_file) as f:
+            CUDA_VERSION = "cu".join(f.readline().strip().split(" ")[-1].split(".")[:2])
+            if CUDA_VERSION not in ["cpu", "cu92", "cu101", "cu102"]:
```
Comment (huangyz0918) [inline setup.py:101]: So how many versions of CUDA do we support now, I saw 92, 101 and 102 here but for tensorflow installation I saw 101, 102 and 110.

File: setup.py (hunk 1 above)
Comment (univerone) [reply setup.py:32]: It's weird that pip installation won't stop by specifying `python_requires`, I'll recover this part of code

File: setup.py (hunk 2 above)
Comment (univerone) [reply setup.py:101]: For torch==1.5.0, only CUDA version 9.2,10.1 and 11.0 can install precompiled GPU packages, for tensorflow==2.3.0, only 10.1, 10.2  and 11.0 are supported(BTW, TensorFlow 2.4.0 seems not working in modelci).

**Comment by HuaizhengZhang** [comment]
LGTM.

File: setup.py [hunk 3]
```diff
@@ -79,17 +74,83 @@ def install_triton_client():
...
+}
+
+CUDA_VERSION = "cpu"
+PYTHON_VER = f'{sys.version_info[0]}{sys.version_info[1]}'
+torch_cuda_version = "cpu"
+tensorflow_cuda_version = "cpu"
+
+
+if "CUDA_HOME" in os.environ:
+    cuda_version_file = os.path.join(os.environ["CUDA_HOME"], "version.txt")
```
Comment (YuanmingLeee) [inline setup.py:114]: a suggestion, `pathlib` is recommended after Python 3.6

File: setup.py [hunk 4]
```diff
@@ -79,17 +74,83 @@ def install_triton_client():
...
+
+            if CUDA_VERSION not in PRECOMPILED_TENSORFLOW_PAIRS:
+                warnings.warn(
+                    f"There is no pre-complied tensorflow-gpu >=2.1.0 with CUDA {CUDA_VERSION}, "
+                    f"and you might need to install tensorflow-gpu >=2.1.0 with CUDA {CUDA_VERSION} from source."
+                )
+            else:
+                tensorflow_cuda_version = CUDA_VERSION
+
+TORCH_INSTALL_URL = f'https://download.pytorch.org/whl/{torch_cuda_version}/torch-1.5.0' \
```
Comment (YuanmingLeee) [inline setup.py:134]: I think Pytorch 1.5.0 do not support Cuda 11.0, you need to check the highest PyTorch support version

Comment (univerone) [reply setup.py:134]: Yes, cuda11.0 will be replaced with CPU version currently, should we use a higher torch version to support cuda11.0?

Comment (YuanmingLeee) [reply setup.py:134]: Better. A100 only supports CUDA >= 11.0

File: setup.py (hunk 2 above)
Comment (YuanmingLeee) [reply setup.py:118]: Can you figure out why TF 2.4.0 not working?

Comment (univerone) [reply setup.py:118]: I came across this issue several times: https://github.com/tensorflow/tensorflow/issues/38012, but 2.4.0 did work this time

**Comment by HuaizhengZhang** [comment]
Is this PR ok? @huangyz0918 @YuanmingLeee 
Have this been well tested? @univerone

**Comment by huangyz0918** [comment]
Does our system support Python 3.9? 

```bash
Processing c:\users\huangyz0918\desktop\ml-model-ci
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'error'
  ERROR: Command errored out with exit status 1:
   command: 'c:\python39\python.exe' 'c:\python39\lib\site-packages\pip\_vendor\pep517\_in_process.py' get_requires_for_build_wheel 'C:\Users\HUANGY~1\AppData\Local\Temp\tmpbfhg5h5u'
       cwd: C:\Users\huangyz0918\AppData\Local\Temp\pip-req-build-9iznuj80
  Complete output (1 lines):
  You are using Python 3.9.0. Python >=3.7.0,<3.9.0 is required.
```

**Comment by HuaizhengZhang** [comment]
I thought we don't need to support python 3.9 currently. 3.7&3.8 is good enough.

We'd better add a badge (support python3.7&3.8)

**Comment by huangyz0918** [comment]
**Tested with success**

Test on conda python 3.7. 3.8 with CUDA 10.2

**Review by huangyz0918 — APPROVED** [review]

**Pull request merged**